src/controllers/clientController.ts

```diff
@@ -121,3 +132,136 @@ export const getClientDetailsByUser = async (
     return res.status(500).json({ message: 'Internal server error' });
   }
 };
+
+//to delete a client
+export const deleteClient = async (
+  req: Request,
+  res: Response
+): Promise<unknown> => {
+  try {
+    const { clientId } = req.body;
+    if (!clientId)
+      return res.status(422).json({ message: 'Client Id is required.' });
+    const user: any = req.user;
+    // fetch client data
+    const client = await prisma.client.findUnique({
+      where: {
+        clientId: clientId
+      }
+    });
+
+    if (!client) res.status(400).json({ message: 'Client not found.' });
+    else if (client && client.userId != user.id)
+      return res
+        .status(404)
+        .json({
+          message:
+            'You do not have the required permissions to delete the client.'
+        });
+
+    //delete client
+    const result = await prisma.client.delete({
+      where: {
+        clientId: clientId
+      }
+    });
+    if (!result) return res.status(400).json({ message: 'Client not found' });
+    return res.status(200).json({ message: 'Client deleted successfully.' });
+  } catch (error) {
+    return res.status(500).json({ message: 'Internal server error' });
+  }
+};
+
+//to update client details
+export const updateClient = async (
+  req: Request,
+  res: Response
+): Promise<unknown> => {
+  try {
+    const { clientId, name, homePageUrl, callbackUrl, description } = req.body;
+    if (!clientId)
+      return res.status(422).json({ message: 'Client Id is required.' });
+    const user: any = req.user;
+    // fetch client data
+    const client = await prisma.client.findUnique({
+      where: {
+        clientId: clientId
+      }
```

Comment:
select{userid:true}
better to make middleware to check if the client exists no?